services/judge/judge_score_cache.go

```diff
@@ -0,0 +1,60 @@
+package judge_service
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	judge_model "github.com/oj-lab/oj-lab-platform/models/judge"
+	gorm_agent "github.com/oj-lab/oj-lab-platform/modules/agent/gorm"
+)
+
+func UpdateScoreCache(ctx context.Context, uid uuid.UUID, verdict judge_model.JudgeVerdict) (*judge_model.ScoreCache, error) {
+	db := gorm_agent.GetDefaultDB()
+	judge, err := judge_model.GetJudge(db, uid)
+	if err != nil {
+		return nil, err
+	}
+	// log_module.AppLogger().WithField("judge", judge).Debug("getjudge")
+	scoreCache, err := judge_model.GetJudgeScoreCache(db, judge.UserAccount, judge.ProblemSlug)
+	if err != nil {
+		// previous empty
+		// log_module.AppLogger().Debug("previous empty")
+
+		scoreCache := judge_model.NewScoreCache(judge.UserAccount, judge.ProblemSlug)
+		if verdict == judge_model.JudgeVerdictAccepted {
+			scoreCache.IsCorrect = true
+			scoreCache.SolveTime = judge.CreateAt
+		}
+		newScoreCache, err := judge_model.CreateJudgeScoreCache(db, scoreCache)
```

Comment:
`CreateJudgeScoreCache` may failed here because of concurrency conflict.
If failed, get data again and continue the update logic.